13.0 refund

cristina-domatix opened the pull request.

cristina-domatix committed:
helpdesk_mgmt: add Return option

We added a Return checkbox in the Teams view inside After sales services.

cristina-domatix committed:
Merge branch '13.0-team_assign' of github.com:Domatix/helpdesk into 13.0-refund

cristina-domatix committed:
helpdesk_mgmt: add button and wizard

We added the Return button, as well as the wizard functionality to select the sale order
of the product we wish to return.

cristina-domatix committed:
helpdesk_mgmt: add stock picking generation

cristina-domatix committed:
helpdesk_mgmt: add return product from ticket

We added a return option for the Ticket in which we can select a stock picking and generate a
reverse transfer for a product.

cristina-domatix committed:
helpdesk_ticket: fix format

cristina-domatix committed:
helpdesk_mgmt: fix conflict in manifest

cristina-domatix committed:
helpdesk_ticket: fix Reverse transfer view

cristina-domatix committed:
Merge branch '13.0' into 13.0-refund

catalin-domatix commented on helpdesk_mgmt/models/stock_picking.py:
```diff
@@ -0,0 +1,5 @@
+
+
+class Picking(models.Model):
+    _inherit = "stock.picking"
```
This is superfluous as you don't do any logic @cristina-domatix

catalin-domatix commented:
This should be another module or part of `helpdesk_sale_order`, not the main module

cristina-domatix committed:
helpdesk_mgmt: move Return to helpdesk_sale_order

cristina-domatix committed:
helpdesk_sale_order: add return product

We added the functionality to return a product when there is a sale order assigned
to it.

cristina-domatix committed:
Merge branch '13.0-refund' of github.com:Domatix/helpdesk into 13.0-refund

cristina-domatix committed:
helpdesk_sale_order: fix Return option to be showed
only when the assign sale order option is also checked.

cristina-domatix committed:
Merge branch '13.0' of github.com:Domatix/helpdesk into 13.0-refund

catalin-domatix commented on helpdesk_mgmt/__manifest__.py:
```diff
@@ -14,7 +14,7 @@
     "Odoo Community Association (OCA)",
     "website": "https://github.com/OCA/helpdesk",
-    "depends": ["mail", "portal", "website"],
+    "depends": ["mail", "portal", "website", "stock"],
```
This should be `sale_stock` as the relation between sale and stock is nonexistent without it, ie: sale order objects do not have the `picking_ids` reference without it.

cristina-domatix committed:
helpdesk_sale_order: fix return option

The product can only be returned when there exists a sale order associated with the ticket.

cristina-domatix committed:
helpdesk_mgmt: fix manifest dependencies

cristina-domatix committed:
helpdesk_sale_order: remove unused variable in test

cristina-domatix committed:
[FIX] helpdesk_sale_order: add mandatory field

The location_id was missing in the wizard view. This field is compulsory to create the Return
transfer, so it triggered a DB error. We made it invisible.

cristina-domatix committed:
Merge branch '13.0' of github.com:Domatix/helpdesk into 13.0-refund

cristina-domatix committed:
helpdesk_sale_order: add link between return transfer and sale order
This happens once validating the generated Return transfer.

cristina-domatix committed:
helpdesk_sale_order: fix style format

catalin-domatix commented on helpdesk_sale_order/wizard/stock_picking_return.py:
```diff
@@ -0,0 +1,49 @@
+        string="Related sale order",
+    )
+
+    linked_transfer_ids = fields.Many2many(
```
should include the reference model's `_name`, ie: `linked_stock_picking_ids` or `linked_picking_ids`

cristina-domatix committed:
[FIX]: rename the Linked Pickings field
